Assets/Scripts/Data/Pickup Pup/Currency/CurrencyData.cs

```diff
@@ -0,0 +1,64 @@
+﻿/*
+ * Author: Grace Barrett-Snyder, Isaiah Mann
+ * Description: Holds the data for a specific form of currency (ex: coins)
+ */
+
+using System.IO;
+using UnityEngine;
+
+[System.Serializable]
+public class CurrencyData
+{
+    #region Instance Accessors
+
+    public int Amount
+    {
+        get
+        {
+            return amount;
+        }
+    }
+
+    public CurrencyType Type
+    {
+        get
+        {
+            return type;
+        }
+    }
+
+    public virtual Sprite Icon
+    {
+        get
+        {
+            return Resources.Load<Sprite>(Path.Combine("Sprites", "Default"));
```

Comment:
Cool, generally isn't a problem. Only needs extra consideration if the superclass contains non-serializable elements, e.g. `Sprite` references. But even then, you can add the Serializable attribute. (And then add `[NonSerialized]` to the class variables Unity / C# do not allow you to serialize)